### PR TITLE
Frost infrastructure wrap-up

### PR DESCRIPTION
**Pull request by rabbitonweb**
In this PR I try to close infrastructure specific changes, so the later on we can mainly focus on plugin development

### CODE REVIEW

File: test/TransformSpec.hs [hunk 1]
```diff
@@ -51,7 +51,8 @@ spec =
                    , CodeBlock ("",["frost:text:insert"],[]) ""]
       let pandoc = Pandoc nullMeta blocks
       -- when
-      let Right(Pandoc _ transformedBlocks) =  run $ runError $ transform [textPlugin "hello world!"] pandoc
+      let Right(Pandoc _ transformedBlocks) =
```
Comment (lukasz-golebiewski) [inline test/TransformSpec.hs]: How is this change related to this commit?

File: src/Frost/PandocRun.hs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import Prelude hiding (readFile, writeFile)
+import Text.Pandoc hiding (trace)
+import Polysemy
+import Polysemy.Input
+import Polysemy.Output
+import Polysemy.Error
+import PolysemyContrib
+
+runInputPandoc :: (
```
Comment (lukasz-golebiewski) [inline src/Frost/PandocRun.hs]: Moving of these functions should be done in a separate commit, because it's now harder to see what changed aside from the names

File: src/PolysemyContrib.hs [hunk 3]
```diff
@@ -17,6 +19,17 @@ data FileProvider m a where
 
 makeSem ''FileProvider
 
+type InMemFileSystem = Map FilePath T.Text
+
+runFileProviderPure :: (Member (State InMemFileSystem) r) => Sem (FileProvider ': r) a -> Sem r a
+runFileProviderPure = interpret $ \case
+  ReadFile path -> do
+    m <- get @InMemFileSystem
```
Comment (lukasz-golebiewski) [inline src/PolysemyContrib.hs]: Does `@InMemFileSystem` come from `Polysemy.State`?

Comment (rabbitonweb) [reply src/PolysemyContrib.hs]: No, it's defined few lines above `type InMemFileSystem = Map FilePath T.Text`

File: src/Frost/PandocRun.hs (hunk 2 above)
Comment (rabbitonweb) [reply src/Frost/PandocRun.hs]: It was move and rename. Nothing else changed.

Comment (rabbitonweb) [reply src/Frost/PandocRun.hs]: Should it be first move (commit 1) then rename (commit 2)?

File: test/TransformSpec.hs (hunk 1 above)
Comment (rabbitonweb) [reply test/TransformSpec.hs]: No formatter in the project yet, so I have to do refactoring by hand :) Could be a separate commit though

**Comment by rabbitonweb** [comment]
Please note this is still WIP

**Comment by rabbitonweb** [comment]
WIP was dropped. This is now ready for review.

**Comment by lukasz-golebiewski** [comment]
I merged it in so you don't have to feel awkward while giving yourself a medal.
https://thecodinglove.com/when-i-merge-my-own-pull-request

**Comment by rabbitonweb** [comment]
:D
